app/services/katello/pulp3/smart_proxy_repository.rb

```diff
@@ -82,6 +82,31 @@ def orphan_repositories
         end
         repo_map
       end
+
+      def repos_in_env_cv(environment = nil, content_view = nil)
+        repos = Katello::Repository
+        repos = repos.in_environment(environment) if environment
+        repos = repos.in_content_views([content_view]) if content_view
+        repo_ids = repos.pluck(:pulp_id)
+        @pulp_repositories.select { |r| repo_ids.include?(r['pulp_id']) }
+      end
+
+      def repos_in_sync_history
+        smart_proxy.smart_proxy_sync_histories.map { |sync_history| sync_history.repository }
+      end
+
+      def current_repositories_data(environment = nil, content_view = nil)
+        @pulp_repositories ||= repos_in_sync_history
+        repo_ids = repos_in_env_cv(environment, content_view)
+        @pulp_repositories.select { |r| repo_ids.include?(r['pulp_id']) }
```

Comment:
i think we can get rid of all this 'pulp_id' comparisons.  This was there before because we were comparing to to results back from pulp2, but since we're just working within our own db, i think we can simplify it quite a bit